### ISSUE TITLE
iSCSI multipath, Can it possibly lead to data inconsistency?

### ISSUE THREAD

**Issue by phucvinh52** [SUSE/lrbd#10]
Hello everybody,
I using lrbd deploy 2 gateways iscsi, 1 client iscsi connect to 2 gateways (multipath gateway A and gateway B, use alua active/enable).
I have a question, when i am writing data to iscsi A, A fail, data continue write to iscsi B. Can it possibly lead to data inconsistency?
Sorry i'm bad english.
Thanks.

**Comment by swiftgist** [comment]
Hello,
  Are you using SUSE kernel for your iSCSI gateways?  The included kernel module handles this.  BTW, upstream has chosen a different path regarding this solution.  I expect the userspace https://github.com/open-iscsi/tcmu-runner to become the preferred solution, but it's currently in progress.  I do plan to adapt lrbd if it is still needed.
  No worries about your English. :)

Eric

**Comment by phucvinh52** [comment]
Hi Eric,
I do not use SUSE kernel for my iSCSI gateways :'(. But thank for your information, It is helpful for me.

**Comment by phucvinh52** [comment]
Hi Eric,
Can I use openSUSE (not SUSE enterprise) for gateway?

**Comment by ddiss** [comment]
Yes, openSUSE Leap 42.2 includes the same kernel. IIUC, www.petasan.org have also picked up the kernel patches.
That said, as Eric mentioned, tcmu-runner will likely become the preferred upstream solution.

**Comment by phucvinh52** [comment]
Thank you very much @ddiss and i have some question but maybe i should open another title.

**Comment by squarejester** [comment]
Isn't it true that using TCMU instead of LRBD has the limitation that is DOES NOT support persistent reservations (like the kind REQUIRED for Windows cluster)? Is so, why is OpenSUSE trying to migrate to TCMU in userspace if it is basically useless for deploying block storage for Windows clusters?

**Comment by ddiss** [comment]
The purpose of migrating to tcmu is to remain consistent with upstream.
The tcmu-runner backend is capable of deferring to the kernel for local reservation handling, which uses a local file for storage of PR state.

**Comment by squarejester** [comment]
In plain English does that mean TCMU instead of lrbd now supports persistent reservation required for Windows clustering or not?
